src/models/tools/geometry/jxg-point.ts

```diff
@@ -27,11 +27,24 @@ const defaultProps = {
         strokeColor: kPointDefaults.strokeColor
       };
 
+// fillColor/strokeColor are ephemeral properties that change with selection;
+// we store the desired colors in clientFillColor/clientStrokeColor for persistence.
+export function syncClientColors(props: any) {
+  const { selectedFillColor, selectedStrokeColor, ...p } = props || {} as any;
+  if (p.fillColor) p.clientFillColor = p.fillColor;
+  if (p.strokeColor) p.clientStrokeColor = p.strokeColor;
```

Comment:
Could you pull these properties out above too?

Reply:
No, because we want to leave them in. Setting `fillColor` is what sets the initial color of the point in JSXGraph. Setting `clientFillColor` is what allows us to reset the color after selection.